lib/lti_1p3/tool/services/ags.ex

```diff
@@ -226,13 +226,22 @@ defmodule Lti_1p3.Tool.Services.AGS do
       %URI{path: line_items_path} = URI.parse(line_items_url)
 
       registration
-      |> Map.get(:auth_server, line_items_url)
+      |> get_line_items_domain(line_items_url)
       |> URI.parse()
       |> Map.put(:path, line_items_path)
       |> URI.to_string()
     end
   end
 
+  defp get_line_items_domain(registration, default) do
+    line_items_service_domain = Map.get(registration, :line_items_service_domain)
+
+    cond do
+      is_nil(line_items_service_domain) or line_items_service_domain == "" -> default
+      true -> line_items_service_domain
+    end
+  end
+
```

Comment:
NIT: you can also write this pattern matching on the function clause:

`
  def get_line_items_domain(%{line_items_service_domain: line_items_service_domain}, _default), do: line_items_service_domain
  def get_line_items_domain(_registration, default), do: default
  `

Reply:
```
def get_line_items_domain(%{line_items_service_domain: nil}, default), do: default
def get_line_items_domain(%{line_items_service_domain: domain}, _default), do: domain
```

Reply:
@eliknebel had the same doubt when implementing, if we should handle the empty string or not. Wanted to just be defensive here in case we define a not null default on that field. 

Anyway with that we are not handling that the key is not present in the registration so handle all the cases we should write something like:
```
def get_line_items_domain(%{line_items_service_domain: nil}, default), do: default
def get_line_items_domain(%{line_items_service_domain: ""}, default), do: default
def get_line_items_domain(%{line_items_service_domain: domain}, _default), do: domain
def get_line_items_domain(_registration, default), do: default
```

Maybe that's clearer and we can quickly change it.

Reply:
I think this works 👍 
We could also merge the first two lines and use a `when` clause

Reply:
Updated. Thanks to all !